### PR TITLE
Added MockData providers for UTs to use

### PR DESCRIPTION
**Pull request by navratan-soni**
Creating Mock Data class for creating push template for Unit tests

## Description

Created data class which provides mocked data for creating push templates, Carousal and Basic.

## Related Issue
https://jira.corp.adobe.com/browse/MOB-21060

## Motivation and Context
For writing UTs for PushTemplate Builder classes and PushTemplate classes, need mocked data (Template objects for various edge cases). This class can further be modified and we can add more mocked data for more tests

## How Has This Been Tested?
These are data classes for UTs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+package com.adobe.marketing.mobile.notificationbuilder.internal.templates
+
+import android.os.Bundle
+import com.adobe.marketing.mobile.notificationbuilder.PushTemplateIntentConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import org.mockito.Mockito
+import org.mockito.kotlin.mock
+
+object MockAEPPushTemplateDataProvider {
+    fun getMockedDataMapWithBasicData(): HashMap<String, String> {
```
Comment (rymorale) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt:21]: nit: can this be returned as a `Map` rather than `HashMap`? our push template classes accept map objects so i don't see a need to use a `HashMap` here

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockDataUtils.kt
Comment (rymorale) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockDataUtils.kt:1]: as the contents of this file are all constants we can just name it `PushTemplateTestConstants` and put all of our testing constants inside

**Comment by rymorale** [comment]
overall, the changes look good to me

**Comment by spoorthipujariadobe** [comment]
@navratan-soni #8 is merged. Could you please refactor this PR based on the changes in #8? Thank you!

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        Mockito.`when`(mockBundle.getString(PushTemplateIntentConstants.IntentKeys.TITLE_TEXT))
+            .thenReturn(MOCKED_TITLE)
+        Mockito.`when`(mockBundle.getString(PushTemplateIntentConstants.IntentKeys.BODY_TEXT))
+            .thenReturn(MOCKED_BODY)
+        Mockito.`when`(mockBundle.getInt(PushTemplateIntentConstants.IntentKeys.PAYLOAD_VERSION))
+            .thenReturn(MOCKED_PAYLOAD_VERSION)
+        return mockBundle
+    }
+
+    fun getMockedBundleWithoutTitle(): Bundle {
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt:42]: This is going to get very tedious, very quickly based on different fields we'll need to remove. Can you build the template with all the keys and have an additional function which takes a list of keys and removes/replaced them?
```
fun removeKeys(val keyToRemove: List<String>) {
// iterate through the list and remove keys from the fully constructed data
}

fun replaceKeys(val keyToReplace: List<String, String>) {
// iterate through the list and replaces keys with values from the list in the fully constructed data
}
```

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt (hunk 1 above)
Comment (navratan-soni) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt]: This is now being kept as MutableMap, which will provide functionality to remove / replace any keys in existing data, for any specific test requirement.
This Mutable map is then passed as parameter to create push template. Kotlin takes care of the conversion from Mutablemap to Map

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockDataUtils.kt
Comment (navratan-soni) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockDataUtils.kt:1]: Added some  utils functions here as well. So we can keep this now.

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt (hunk 2 above)
Comment (navratan-soni) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt]: Added the suggested functionality. Thanks !

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+import android.os.Bundle
+import com.adobe.marketing.mobile.notificationbuilder.internal.NotificationPriority
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateType
+import org.mockito.Mockito
+import org.mockito.kotlin.mock
+
+object MockAEPPushTemplateDataProvider {
+    fun getMockedDataMapWithBasicData(): MutableMap<String, String> {
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt:22]: nit: can we call this `getMockedDataMapWithRequiredData`?

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        return mutableMapOf(
+            Pair(PushTemplateConstants.PushPayloadKeys.TITLE, MOCKED_TITLE),
+            Pair(PushTemplateConstants.PushPayloadKeys.BODY, MOCKED_BODY),
+            Pair(PushTemplateConstants.PushPayloadKeys.VERSION, MOCKED_PAYLOAD_VERSION)
+        )
+    }
+    /**
+     * Returns a mocked data bundle with basic data.
+     */
+    fun getMockedBundleWithMinimalData(): Bundle {
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt:32]: nit: can we call this `getMockedBundleWithRequiredData`?

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+import com.adobe.marketing.mobile.notificationbuilder.internal.NotificationPriority
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateType
+import org.mockito.Mockito
+import org.mockito.kotlin.mock
+
+object MockAEPPushTemplateDataProvider {
+    fun getMockedDataMapWithBasicData(): MutableMap<String, String> {
+        return mutableMapOf(
+            Pair(PushTemplateConstants.PushPayloadKeys.TITLE, MOCKED_TITLE),
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt:24]: Why do we need `Pair` here?

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            .thenReturn(MOCKED_PAYLOAD_VERSION)
+        return mockBundle
+    }
+
+    fun getMockedAEPDataMapWithAllKeys(): MutableMap<String, String> {
+        return mutableMapOf(
+            PushTemplateConstants.PushPayloadKeys.TAG to MOCKED_TAG,
+            PushTemplateConstants.PushPayloadKeys.TEMPLATE_TYPE to PushTemplateType.BASIC.value,
+            PushTemplateConstants.PushPayloadKeys.ACTION_URI to MOCKED_ACTION_URI,
+            PushTemplateConstants.PushPayloadKeys.ACTION_TYPE to "NONE",
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt:48]: nit: We can use `PushTemplateConstants.ActionType.NONE.name` here

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    fun getMockedAEPBundleWithAllKeys(): Bundle {
+        val mockBundle = mock<Bundle>()
+        Mockito.`when`(mockBundle.getString(PushTemplateConstants.PushPayloadKeys.TAG))
+            .thenReturn(MOCKED_TAG)
+        Mockito.`when`(mockBundle.getString(PushTemplateConstants.PushPayloadKeys.TEMPLATE_TYPE))
+            .thenReturn(PushTemplateType.BASIC.value)
+        Mockito.`when`(mockBundle.getString(PushTemplateConstants.PushPayloadKeys.ACTION_URI))
+            .thenReturn(MOCKED_ACTION_URI)
+        Mockito.`when`(mockBundle.getString(PushTemplateConstants.PushPayloadKeys.ACTION_TYPE))
+            .thenReturn("NONE")
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt:80]: nit: We can use `PushTemplateConstants.ActionType.NONE.name` here

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockDataUtils.kt [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    }
+}
+
+fun <K, V> MutableMap<K, V>.replaceValueInMap(vararg keyValues: Pair<K, V>) {
+    for ((key, value) in keyValues) {
+        this[key] = value
+    }
+}
+
+internal fun provideMockedBasicPushTemplateWithMinimalData(isFromIntent: Boolean = false): BasicPushTemplate {
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockDataUtils.kt:63]: nit: `provideMockedBasicPushTemplateWithRequiredData`

**Review by spoorthipujariadobe — APPROVED** [review]
LGTM, just a few nits around naming. Great job on the first PR!

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt (hunk 3 above)
Comment (navratan-soni) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt]: Done

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt (hunk 5 above)
Comment (navratan-soni) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt]: Done

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt (hunk 7 above)
Comment (navratan-soni) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt]: DOne

**Review by PratthamArora — APPROVED** [review]
